src/sensai/vector_model.py

```diff
@@ -376,7 +389,7 @@ def convertClassProbabilitiesToPredictions(self, df: pd.DataFrame):
         """
         modelOutputVariableName = self.getModelOutputVariableNames()[0]
         y = self._convertClassProbabilitiesToPredictions(df, predictedVariableName=modelOutputVariableName)
-        # TODO check sanity
+        # TODO check sanity: where should we be applying post-processing?
```

Comment:
The way it is now, post-processing can only happen after the argmax operation - the invertible target transformer can convert between labels and back. We can't feed probability vectors to it

Reply:
If you look at the implementation of `VectorModel.predict`, you'll find that it's being applied at the very end, after having called `_predict`, which, for classification models, now (indirectly) calls `_applyPostProcessing` again in `convertClassProbabilitiesToPredictions`.
I see a problem there. ;)

Anyway, this will be my last comment. I won't have time to take this any further. If you want to, take over.